api/v1/nodescan/service.go

```diff
@@ -238,16 +240,35 @@ func (s *serviceImpl) getRuntimeVulns(containerRuntime *v1.GetNodeVulnerabilitie
 	return nil, nil
 }
 
-func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
-	if stringutils.AtLeastOneEmpty(req.GetKernelVersion(), req.GetOsImage()) {
-		return nil, status.Error(codes.InvalidArgument, "both os image and kernel version are required")
+func (s *serviceImpl) GetNodeVulnerabilities(ctx context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+	resp := &v1.GetNodeVulnerabilitiesResponse{
+		ScannerVersion: s.version,
+	}
+	// If NodeInventory is empty `req.GetComponents() == nil` then fallback to v1 scanning
+	if req.GetComponents() == nil || !featureFlags.RHCOSNodeScanning.Enabled() {
+		return s.getNodeVulnerabilitiesLegacy(ctx, req, resp)
+	}
+
+	if !wellknownnamespaces.IsRHCOSNamespace(req.GetComponents().GetNamespace()) {
+		// Non-RHCOS system with node inventory detected!
+		// We can provide a list of pkgs but cannot scan them -> setting a note to inform the user.
+		resp.NodeNotes = append(resp.GetNodeNotes(), v1.NodeNote_NODE_UNSUPPORTED)
```

Comment:
Should we add the same note for inventories in certified RHEL namespaces (including RHCOS) but without content sets?

Reply:
Like this? [a5f7af3](https://github.com/stackrox/scanner/pull/1065/commits/a5f7af3b570a8b84564da9244ec7a9d373a124c8)